common/lib.rs

```diff
@@ -17,7 +17,7 @@ pub fn hash<T: Hash>(t: &T) -> i32 {
     s.finish() as i32
 }
 
-pub fn eq<T: Eq>(t: &T, other: &T) -> bool {
+pub fn eq<T: PartialEq>(t: &T, other: &T) -> bool {
```

Comment:
This was changed because `Jwt` doesn't currently implement `Eq`. We could probably add it, but it was a stronger requirement than Java `equals` needed regardless. `PartialEq` already requires that the equality is symmetric and transitive.